src/components/Compose.js

```diff
@@ -51,8 +129,30 @@ function Compose(props) {
         return;
       }
 
+      // if the user selects a different identity, the focus will be on the select
+      // this is used to allow up/down to cycle through identities if the focus is on select
+      const isSelect = e.target.nodeName === 'SELECT';
+
+      const isUpOrDown = e.code === 'ArrowUp' || e.code === 'ArrowDown';
+
       if (input.current && !isPickerVisible) {
-        input.current.focus();
+        if (!isSelect) {
+          const areMessages = usersRecentMessages.length > 0;
+
+          // so only the up key starts the cycle through recent messages
+          const keyDownNoMessage = (!message && e.code === 'ArrowDown');
+
+          if (areMessages &&  isUpOrDown && !isSelect && !keyDownNoMessage) {
+            const nextMessage = getNextMessage(e.code);
+
+            if(nextMessage !== null && nextMessage !== message) {
+              setMessage(nextMessage);
+            }
+          }
+          input.current.focus();
+        } else if(isSelect && !isUpOrDown) { // so non up/down keys bring focus to textarea
+          input.current.focus();
+        }
       }
     };
     document.addEventListener('keydown', onKeyDown);
```

Comment:
I'm slightly concerned about this having a global effect. I think it would be better if the `keydown` for history lookup would be listened only on the `<input/>` itself, and not for the whole document